### PR TITLE
Token refactor

### LINKED ISSUE #24: Access code (BE)
As a user, when I want to create a new event, I first need to enter in an access code. If it's the correct code, I am redirected to the create from. If it's incorrect, I am notified the access code is incorrect. 

- [x] Create an authenticate method
- [ ] Run the authenticate method before creation 
- [ ] Require access token in create params

### PR DESCRIPTION
**Pull request by TravisBorgsmiller**
## What functionality does this accomplish?
Helps make token functionality more secure with Phoenix. Token encryption and allows for token creation

### Test Coverage Percentage: 78.3%
- [x] No tests have been changed
- [ ] Some tests have been changed:
- [ ] All of the tests have been changed:
## Checklist:
- [x] I have reviewed my code
- [ ] I have annotated or commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code
- [ ] I have partially tested my code (please explain why):
## Related Issues:
Resolves #24 

## Helpful Resources:
https://hexdocs.pm/phoenix/0.16.0/Phoenix.Token.html

### CODE REVIEW

File: lib/drag_time_api_web/router.ex [hunk 1]
```diff
@@ -11,7 +11,7 @@ defmodule DragTimeApiWeb.Router do
     resources "/locations", LocationController, except: [:new, :edit]
     resources "/events", EventController, except: [:new, :edit]
     resources "/artists", ArtistController, except: [:new, :edit]
-    resources "/access", AccessController, execpt: [:new, :edit]
+    resources "/access", AccessController, only: [:index, :create]
```
Comment (javier-aguilar) [inline lib/drag_time_api_web/router.ex:14]: 👍

File: lib/drag_time_api_web/controllers/access_controller.ex [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+  def create(conn, %{"access" => access_code_params}) do
+    code = Phoenix.Token.sign(secret, "access", access_code_params)
+    with {:ok, %Access{} = access} <- Accesses.create_access(%{"access_code" => code}) do
+      conn
+      |> put_status(:created)
+    end
+    send_resp(conn, 200, '{"success" : #{code}}')
+  end
+
+  defp secret do
```
Comment (javier-aguilar) [inline lib/drag_time_api_web/controllers/access_controller.ex:29]: What is this doing? Does this not cause an error due to "defp"

File: config/test.exs [hunk 3]
```diff
@@ -20,3 +20,6 @@ config :drag_time_api, DragTimeApiWeb.Endpoint,
 
 # Print only warnings and errors during test
 config :logger, level: :warn
+
+# Bcrypt configuration 
+config :bcrypt_elixir, :log_rounds, 4
```
Comment (javier-aguilar) [inline config/test.exs]: Nice

**Review by collinkallery — APPROVED** [review]
Lookin good! Thanks Travis

File: lib/drag_time_api_web/controllers/access_controller.ex (hunk 2 above)
Comment (TravisBorgsmiller) [reply lib/drag_time_api_web/controllers/access_controller.ex:29]: The p makes this method private

File: config/test.exs (hunk 3 above)
Comment (TravisBorgsmiller) [reply config/test.exs]: I should actually take this off now since I used Phoenix's internal token tool instead of bcrypt

**Review by javier-aguilar — APPROVED** [review]

**Pull request merged**